From 099072dd7aa3a39acd7a479eda3423052c90969b Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Tue, 14 Sep 2010 07:44:01 -0500
Subject: [PATCH 29/32] v4l2: prefer NV12

All else being equal, the buffer sizes are smaller compared to a 422 format
like YUY2/UYVY..  although ideally rank would come from driver.
---
 sys/v4l2/gstv4l2object.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 606ec21..be33584 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -910,6 +910,9 @@ gst_v4l2_object_format_get_rank (const struct v4l2_fmtdesc *fmt)
       break;
 
     case V4L2_PIX_FMT_NV12:    /* 12  Y/CbCr 4:2:0  */
+      rank = YUV_BASE_RANK + 11;
+      break;
+
     case V4L2_PIX_FMT_NV21:    /* 12  Y/CrCb 4:2:0  */
     case V4L2_PIX_FMT_YYUV:    /* 16  YUV 4:2:2     */
     case V4L2_PIX_FMT_HI240:   /*  8  8-bit color   */
-- 
1.7.1

